[DOCS] Adds descriptions for Enrich APIs

szabosteve opened the pull request:
## Overview

Related to https://github.com/elastic/elasticsearch-specification/issues/964.

Adds descriptions to Enrich API properties. Source of info: https://www.elastic.co/guide/en/elasticsearch/reference/master/enrich-apis.html

abdonpijpelink commented on specification/enrich/execute_policy/ExecuteEnrichPolicyRequest.ts:
```diff
@@ -21,15 +21,34 @@ import { RequestBase } from '@_types/Base'
 import { Name } from '@_types/common'
 
 /**
+ * Creates the enrich index for an existing enrich policy.
+ * The enrich index contains documents from the policy’s source indices. Enrich
+ * indices always begin with `.enrich-*`, are read-only, and are force merged.
+ * Enrich indices should be used by the enrich processor only. Avoid using
+ * enrich indices for other purposes.
+ * Once created, you cannot update or index documents to an enrich index.
+ * Instead, update your source indices and execute the enrich policy again. This
+ * creates a new enrich index from your updated source indices. The previous
+ * enrich index will deleted with a delayed maintenance job. By default this is
+ * done every 15 minutes.
+ * Because this API request performs several operations, it may take a while to
+ * return a response.
```
Maybe this is too much information for reference docs? I'd stick to just _"Creates the enrich index for an existing enrich policy."_ here, and maybe add a `@doc_id` for more information.

abdonpijpelink reviewed (approved):
LGTM. Left a few non-blocking comments.

szabosteve commented:
@abdonpijpelink Thanks for your review. I addressed your comments via [b2361b7](https://github.com/elastic/elasticsearch-specification/pull/2215/commits/b2361b72cf6c4eb6dc254871d27e4b35da052100).
